test_plans/oneapi/ext_oneapi_current_device.asciidoc

```diff
@@ -0,0 +1,44 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for ext oneapi current device
+
+This is a test plan for the device selector as described in SYCL 2020 revision 9 specification.
+
+Estimated development time is one day.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All of the tests described below are performed only on the default device that
+can be selected on the CTS command line.
+
+== Tests
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_CURRENT_DEVICE` so they can be skipped
+if feature is not supported.
+
+=== Function `get_current_device`
+
+1. Positive case. This test can be performed by seting default device, calling from a host thread the function sycl::ext::oneapi::experimental::this_thread::get_current_device 
+and compare function output with the default device which was set earlier, check thread ID 
+
+2. Negative case 1. Implement the same as positive but call function from kernel(not host) code. Use XFAIL to detect compilation error
+
+3. Negative case 2. Create async task and call the function sycl::ext::oneapi::experimental::this_thread::get_current_device from async task, Use XFAIL to detect compilation error.
```

Comment:
What does "async task" mean?  Is this a SYCL host task?

Assuming you mean host task, we should not test this case.  There is no requirement for the compiler to diagnose a compile-time error.  In fact, I don't think DPC++ would generate a compilation error in this case because we can't tell at compile time whether the call comes from a host task.